File: iteration2/src/utils/AuthenticateService.java
package iteration2.src.utils;

import java.util.ArrayList;
import java.util.List;

import iteration2.src.models.User;

public class AuthenticateService {

    public User authenticateUser(String userID, String password) {

        DatabaseManager database = DatabaseManager.getInstance();
        List<User> users = new ArrayList<>();

        users.addAll(database.getAdmins());
        users.addAll(database.getAdvisors());
        users.addAll(database.getStudents());

        for (User user : users) {
            if (user.checkCredentials(userID, password)) {
                Util.getLogger().info("User:" + userID + " logged in");
                return user;

            }

        }
        Util.getLogger().info("User:" + userID + " failed to log in");
        return null;

    }

}

File: iteration3/src/utils/AuthenticateService.py
import logging
from utils.DatabaseManager import DatabaseManager
from models.User import User

class AuthenticateService(object):
    
    def authenticateUser(self, userID, password):
        
        databaseManager = DatabaseManager.getInstance()
        users: [User] = []
        users.extend(databaseManager.getAdminList())
        users.extend(databaseManager.getAdvisorList())
        users.extend(databaseManager.getStudentList())

        for user in users:
              if user.checkCredentials(userID, password):
                   logging.log(logging.INFO, "User: "+ userID +" logged in")
                   return user
        
        logging.log(logging.WARNING, "User: "+ userID +" faild to log in")
        return None
